external-crates/move/crates/move-compiler/src/expansion/ast.rs

```diff
@@ -965,6 +965,92 @@ impl std::fmt::Display for Value_ {
     }
 }
 
+//**************************************************************************************************
+// Shared
+//**************************************************************************************************
+
+// Implicit aliases for the Move Stdlib:
+// use std::vector;
+// use std::option::{Self, Option};
+pub const IMPLICIT_STD_MODULES: &[Symbol] = &[symbol!("option"), symbol!("vector")];
+pub const IMPLICIT_STD_MEMBERS: &[(Symbol, Symbol, ModuleMemberKind)] = &[(
+    symbol!("option"),
+    symbol!("Option"),
+    ModuleMemberKind::Struct,
+)];
+
+// Implicit aliases for Sui mode:
+// use sui::object::{Self, ID, UID};
+// use sui::transfer;
+// use sui::tx_context::{Self, TxContext};
+pub const IMPLICIT_SUI_MODULES: &[Symbol] = &[
+    symbol!("object"),
+    symbol!("transfer"),
+    symbol!("tx_context"),
+];
+pub const IMPLICIT_SUI_MEMBERS: &[(Symbol, Symbol, ModuleMemberKind)] = &[
+    (symbol!("object"), symbol!("ID"), ModuleMemberKind::Struct),
+    (symbol!("object"), symbol!("UID"), ModuleMemberKind::Struct),
+    (
+        symbol!("tx_context"),
+        symbol!("TxContext"),
+        ModuleMemberKind::Struct,
+    ),
+];
+
+#[derive(Copy, Clone, Debug)]
+pub enum ModuleMemberKind {
+    Constant,
+    Function,
+    Struct,
+    Enum,
+}
+
+impl ModuleMemberKind {
+    pub fn case(self) -> NameCase {
+        match self {
+            ModuleMemberKind::Constant => NameCase::Constant,
+            ModuleMemberKind::Function => NameCase::Function,
+            ModuleMemberKind::Struct => NameCase::Struct,
+            ModuleMemberKind::Enum => NameCase::Enum,
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum NameCase {
+    Constant,
+    Function,
+    Struct,
+    Enum,
+    Module,
+    ModuleMemberAlias(ModuleMemberKind),
+    ModuleAlias,
+    Variable,
+    Address,
+    TypeParameter,
+}
+
+impl NameCase {
+    pub const fn name(&self) -> &'static str {
+        match self {
+            NameCase::Constant => "constant",
+            NameCase::Function => "function",
+            NameCase::Struct => "struct",
+            NameCase::Enum => "enum",
+            NameCase::Module => "module",
+            NameCase::ModuleMemberAlias(ModuleMemberKind::Function) => "function alias",
+            NameCase::ModuleMemberAlias(ModuleMemberKind::Constant) => "constant alias",
+            NameCase::ModuleMemberAlias(ModuleMemberKind::Struct) => "struct alias",
+            NameCase::ModuleMemberAlias(ModuleMemberKind::Enum) => "enum alias",
+            NameCase::ModuleAlias => "module alias",
+            NameCase::Variable => "variable",
+            NameCase::Address => "address",
+            NameCase::TypeParameter => "type parameter",
+        }
+    }
+}
```

Comment:
I'm not sure if these should live in the AST. It might be worth making a new module called `name_validation` that has these plus the checking functions. Then at least all the code and definitions will still live together.

Reply:
+1 to these not being in the AST. You could stick them in `expansion/mod.rs` if you are really struggling for a place. But something like `name_validation` with all of the other various checking names (some of which might have been dragged into the AST already?) sounds like a great long term solution

Reply:
Done, though it required a bit more massaging as all these name validation functions take `context` as an argument to generate diagnostics